src/crawlee/http_clients/_httpx.py

```diff
@@ -85,6 +87,7 @@ def __init__(
         ignore_http_error_status_codes: Iterable[int] = (),
         http1: bool = True,
         http2: bool = True,
+        append_common_headers: bool = True,
```

Comment:
I'd like to discuss making a header generator interface and passing it via constructor, instead of tying one specific implementation to the value of this parameter. What do you think?

Reply:
good idea, thanks